package.json

```diff
@@ -85,6 +86,7 @@
     "format": "remark . -qfo && prettier . --write && xo --fix",
     "test-api": "node test",
     "test-coverage": "nyc --reporter lcov tape test/index.js",
+    "test-fuzz": "jsfuzz fuzzer.js corpus",
```

Comment:
:notebook: the fuzzer can run forever, it's probably best not to run the fuzzer itself on CI.

However, the corpus it generates could potentially be used for testing.
Including the corpus for testing is optional, in this PR I did not include any of the generated test cases (yet).

Reply:
It seems that the fuzzer can explore up to around 90% coverage in an hour or two.
It may be worth exploring the possibility of occasionally running the fuzzer for a couple hours the shutting it down based off a timer, to explore edge cases potentially not covered in the unit tests.

Reply:
Travis seems to have a run time of 50 minutes tops. Could we perhaps have this in a separate repo (micromark/fuzzcheck or so), with a weekly/monthly travis cronjob? 🤔

Reply:
Are there other things like this we could (automatically) test for?

Reply:
:thinking: a few other things:
* [Symbolic Execution of the code](https://en.wikipedia.org/wiki/Symbolic_execution) (example [1](https://github.com/Samsung/jalangi2), [2](https://github.com/ExpoSEJS/ExpoSE)), like fuzzing, SymEx can be very time consuming.
* [More ESLint rules](https://github.com/dustinspecker/awesome-eslint#code-quality), like [sonar](https://github.com/SonarSource/eslint-plugin-sonarjs), [unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn), [node](https://github.com/mysticatea/eslint-plugin-node), and [regex](https://github.com/BrainMaestro/eslint-plugin-optimize-regex)

Reply:
> Travis seems to have a run time of 50 minutes tops. Could we perhaps have this in a separate repo (micromark/fuzzcheck or so), with a weekly/monthly travis cronjob?

That's one option, another would be to leverage GithHub actions, which have a limit of 6 hours.
https://docs.github.com/en/actions/reference/usage-limits-billing-and-administration#usage-limits

Reply:
damn that’s long 😅 well. All for doing that. I think that having it in some cron job will help because otherwise we’ll forget. But on the other hand: someone has to check when the cron job fails. Does GH CI have cron jobs, and some notification if it fails?

lint: I’m all for more rules, although a) xo is very stylistic (to the point of being wrong when doing such lowlevel things), b) I think coverage will have checked sonar and regex. But I’m all for trying ’em out!

CST + fuzz: ah, yeah, that would be good! I’m now working on the AST, and there’s just like: so many ways for markdown to accidentally parse differently. Same with HTML: it also can’t round trip 100%. I think that’s fine for the AST. Although it makes sense to me that the CST should roundtrip 100%.
To illustrate the AST: GFM parses `a.b-c_d@a.b` as an email link. mdast-util-to-markdown serializes that as `<a.b-c_d@a.b>`, so that it is also a link to non-GFM. However, CM has slightly different email parsing because `a` is too short for a domain name so it bails. The only way to properly address these edge cases is to serialize, use a markdown parser to parse the result, and see if it’s indeed the same as the origin, and that’s just too much work.

Reply:
> But on the other hand: someone has to check when the cron job fails. Does GH CI have cron jobs, and some notification if it fails?

it does support scheduled/cron jobs https://docs.github.com/en/actions/reference/events-that-trigger-workflows#schedule
for notifications, I'm not sure what the default behavior is, but there are actions to sent emails, slack messages, etc.

Reply:
ok, let’s first focus on getting this in, then figure out where we’re running cron jobs!